src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -379,85 +488,116 @@ void Snippet::initOptimalPrimitiveDescriptor() {
-#ifdef SNIPPETS_LIBXSMM_TPP
-    const auto& lir = snippetAttrs.snippet->convert_body_to_linear_ir(static_cast<size_t>(parallel_get_max_threads()), 256,
-                                                                      std::make_shared<snippets::CPUShapeInferSnippetsFactory>());
-    lir->set_loop_depth(std::min(2ul, lir->get_master_shape().size()));
+std::pair<Subgraph::ControlFlowConfig, Subgraph::ControlFlowPasses> Subgraph::getControlFlowPasses() const {
+    ControlFlowConfig lowering_config = std::make_shared<ov::snippets::lowered::pass::PassConfig>();
+    ControlFlowPasses backend_passes;
+
+#if defined(OPENVINO_ARCH_X86_64)
+    using PassPosition = ov::snippets::pass::PassPosition;
+    using Place = PassPosition::Place;
+#   define SNIPPETS_REGISTER_PASS_RELATIVE(PASS_PLACE, TARGET_PASS, PASS, ...) \
+            backend_passes.emplace_back(PassPosition(PASS_PLACE, TARGET_PASS::get_type_info_static()), std::make_shared<PASS>(__VA_ARGS__))
 #else
-    snippetAttrs.snippet->convert_body_to_linear_ir(static_cast<size_t>(parallel_get_max_threads()), 256,
-                                                    std::make_shared<snippets::CPUShapeInferSnippetsFactory>());
+#    define SNIPPETS_REGISTER_PASS_RELATIVE(PASS_PLACE, TARGET_PASS, PASS, ...)
+#endif  // OPENVINO_ARCH_X86_64
+
+    SNIPPETS_REGISTER_PASS_RELATIVE(Place::After, ov::snippets::lowered::pass::MarkLoops,
+                                    ov::intel_cpu::pass::BrgemmBlocking);
+    SNIPPETS_REGISTER_PASS_RELATIVE(Place::After, ov::snippets::lowered::pass::InsertLoops,
+                                    ov::intel_cpu::pass::FuseLoadStoreConvert);
+    SNIPPETS_REGISTER_PASS_RELATIVE(Place::After, ov::intel_cpu::pass::FuseLoadStoreConvert,
+                                    ov::intel_cpu::pass::SetBrgemmCopyBBuffersShape);
+
+#ifdef SNIPPETS_LIBXSMM_TPP
+    // Note: temporary disabled. Re-enable after ticket 132833 is resolved
+    lowering_config->disable<ov::snippets::lowered::pass::OptimizeDomain>();
+    SNIPPETS_REGISTER_PASS_RELATIVE(Place::After, ov::intel_cpu::pass::FuseLoadStoreConvert,
+                                    ov::intel_cpu::tpp::pass::SetTPPLeadingDim);
 #endif
+
+#undef SNIPPETS_REGISTER_PASS_RELATIVE
+    return std::make_pair(lowering_config, backend_passes);
 }
 
-ov::element::Type Snippet::getRuntimePrecision() const {
-    std::vector<ov::element::Type> inputPrecisions;
-    for (size_t i = 0; i < getParentEdges().size(); i++) {
-        auto parentEdge = getParentEdgeAt(i);
-        if (parentEdge && parentEdge->getStatus() == Edge::Status::Validated && !parentEdge->getParent()->isConstant()) {
-            inputPrecisions.emplace_back(DnnlExtensionUtils::DataTypeToElementType((parentEdge->getMemoryPtr()->getDataType())));
-        }
+uint8_t Subgraph::getBroadcastingMask(const std::vector<VectorDims>& input_shapes) {
+    uint8_t mask = 0;
+    // TODO: add check for non-eltwise inputs
```

Comment:
We at least should check `has_domain_sensitive_ops` here

Reply:
Works for now, but we should think about creating a set of predefined "special" masks in future